jenkins-scripts/docker/reprepro_updater.bash

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash -x
+
+set -e
+
+# Knowing Script dir beware of symlink
+[[ -L "${0}" ]] && SCRIPT_DIR=$(readlink "${0}") || SCRIPT_DIR="${0}"
+SCRIPT_DIR="${SCRIPT_DIR%/*}"
+
+PYTHON_VENV="${WORKSPACE}/venv"
+REPREPRO_GIT_REPO="https://github.com/j-rivero/reprepro-updater"
+REPREPRO_GIT_BRANCH="${REPREPRO_GIT_BRANCH:-refactor}"
+REPREPRO_REPO_PATH="$WORKSPACE/repo"
+
+REPREPRO_PARAMS=''
+if ${COMMIT}; then
+    REPREPRO_PARAMS='--commit'
+fi
+
+if [[ ! -f "${HOME}/.buildfarm/reprepro-updater.ini" ]]; then
+    echo "The reprepro configuration file was not found"
+    exit -1
+fi
+
+echo '#BEGIN: prepare and join python-venv'
+if [[ ! -d ${PYTHON_VENV} ]]; then
+  python3 -m venv "${PYTHON_VENV}"
+  source "${PYTHON_VENV}/bin/activate"
+  pip3 install configparser
+  # for some reason the build is broken on Xenial but seems to install
+  # the necessary support
+  pip3 install pyyaml || true
+else
+  source "${PYTHON_VENV}/bin/activate"
+fi
+echo '# END SECTION'
+
+echo '# BEGIN SECTION: clone the git repo'
+rm -fr ${REPREPRO_REPO_PATH}
+git clone "${REPREPRO_GIT_REPO}" -b "${REPREPRO_GIT_BRANCH}" "${REPREPRO_REPO_PATH}"
+echo '# END SECTION'
+
+export PYTHONPATH="${REPREPRO_REPO_PATH}/src"
+export GNUPGHOME=/var/lib/jenkins/.gnupg 
+
+echo '# BEGIN SECTION: run reprepro'
+cd "${REPREPRO_REPO_PATH}/scripts"
+sudo -E bash -c "PYTHONPATH=${REPREPRO_REPO_PATH}/src python3 import_upstream.py ${REPREPRO_PARAMS} \
```

Comment:
Why is reprepro-updater being run as root here?

Reply:
This comes from the very bad habit of the old server where root was in charge of managing the repository. That configuration did not change while migrating to the new packages machine. Should be changed but there is a good number of places where that needs to happen so a bit of planned work is need to make it happen.